src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -131,16 +132,26 @@ void CsvImportWidget::updateTableview() {
 
 void CsvImportWidget::updatePreview() {
 
+    int minSkip = 0;
+    if (m_ui->checkBoxFieldNames->isChecked())
+        minSkip = 1;
     m_ui->labelSizeRowsCols->setText(m_parserModel->getFileInfo());
-    m_ui->spinBoxSkip->setValue(0);
-    m_ui->spinBoxSkip->setMaximum(qMax(0, m_parserModel->rowCount() - 1));
+    m_ui->spinBoxSkip->setRange(minSkip, qMax(minSkip, m_parserModel->rowCount() - 1));
+    m_ui->spinBoxSkip->setValue(minSkip);
 
-    int i;
+    int i, emptyId = 0;
+    QString columnName;
```

Comment:
Well, that `i` variable is used some lines below so it needs to be at that scope level.